lib/seen_subject_remover.rb

```diff
@@ -20,44 +19,17 @@ def append_ids_size
   end
 
   def seen_before_set
-    @seen_before ||= if user_seen_subject
-                       SetMemberSubject.where(id: append_ids)
-                       .joins(:subject)
-                       .where(subjects: { id: user_seen_subject.subject_ids })
-                     else
-                       SetMemberSubject.none
-                     end.pluck(:id)
+    @seen_before ||=
+      if user_seen_subject
+        SetMemberSubject.where(id: append_ids)
+          .joins(:subject)
+          .where(subjects: { id: user_seen_subject.subject_ids })
```

Comment:
Apart from ensuring we haven't deleted the subject independent of the sms, no. I can't see a compelling reason for this though, i'll update.